impeller/display_list/canvas.cc

```diff
@@ -1081,28 +971,38 @@ void Canvas::SaveLayer(const Paint& paint,
 
     std::shared_ptr<Texture> input_texture;
 
-    // If the backdrop ID is not the no-op id, and there is more than one usage
+    // If the backdrop ID is not nullopt and there is more than one usage
     // of it in the current scene, cache the backdrop texture and remove it from
     // the current entity pass flip.
     bool will_cache_backdrop_texture = false;
     BackdropData* backdrop_data = nullptr;
+    size_t backdrop_count = 1;
     if (backdrop_id.has_value()) {
       std::unordered_map<int64_t, BackdropData>::iterator backdrop_data_it =
           backdrop_data_.find(backdrop_id.value());
       if (backdrop_data_it != backdrop_data_.end()) {
         backdrop_data = &backdrop_data_it->second;
         will_cache_backdrop_texture =
             backdrop_data_it->second.backdrop_count > 1;
+        backdrop_count = backdrop_data_it->second.backdrop_count;
       }
     }
 
     if (!will_cache_backdrop_texture ||
         (will_cache_backdrop_texture && !backdrop_data->texture_slot)) {
-      input_texture = FlipBackdrop(render_passes_,              //
-                                   GetGlobalPassPosition(),     //
-                                   clip_coverage_stack_,        //
-                                   renderer_,                   //
-                                   will_cache_backdrop_texture  //
+      backdrop_count_ -= backdrop_count;
+
+      // The onscreen texture can be flipped to if:
+      // 1. The device supports framebuffer fetch
+      // 2. There are no more backdrop filters
+      // 3. The current render pass is for the onscreen pass.
```

Comment:
3. is important because the last backdrop filter may happen in a saveLayer. At that point, we already "missed" the chance to restore to the onscreen.

To fix this we would need to track the backdrop filters per layer, but that is harder to due upfront as layers may be combined.